Could the test failures be related?

```python
FAILED tests/test_interactiveshell.py::TestShowTracebackAttack::test_set_show_tracebacks_none - TypeError: 'NoneType' object is not callable
FAILED tests/test_magic.py::test_notebook_export_json_with_output - AssertionError: Outputs do not match for cell 2 with source "print('test')"
assert [{'name': 'st...esttest\n\n"}] == [{'name': 'st...t': 'test\n'}]
  At index 0 diff: {'name': 'stdout', 'output_type': 'stream', 'text': "\ndefaultdict(<class 'list'>, {})\n\ntesttest\n\n"} != {'name': 'stdout', 'output_type': 'stream', 'text': 'test\n'}
  Full diff:
    [
     {'name': 'stdout',
      'output_type': 'stream',
  -   'text': 'test\n'},
  ?            ----   --
  +   'text': '\n'
  +           "defaultdict(<class 'list'>, {})\n"
  +           '\n'
  +           'testtest\n'
  +           '\n'},
    ]
```

All tests pass on `main` branch:

![image](https://github.com/user-attachments/assets/c94158a1-8a18-49dd-b568-6b3891c88481)

--- a/IPython/core/interactiveshell.py
+++ b/IPython/core/interactiveshell.py
@@ -3152,14 +3152,17 @@ class InteractiveShell(SingletonConfigurable):
         try:
             result = runner(coro)
         except BaseException as e:
-            info = ExecutionInfo(
-                raw_cell, store_history, silent, shell_futures, cell_id
-            )
-            result = ExecutionResult(info)
-            result.error_in_exec = e
-            self.showtraceback(running_compiled_code=True)
-        finally:
-            return result
+            try:
+                info = ExecutionInfo(
+                    raw_cell, store_history, silent, shell_futures, cell_id
+                )
+                result = ExecutionResult(info)
+                result.error_in_exec = e
+                self.showtraceback(running_compiled_code=True)
+            except:
+                pass
+
+        return result
 
     def should_run_async(
         self, raw_cell: str, *, transformed_cell=None, preprocessing_exc_tuple=None

